File: 350-intersection-of-two-arrays-ii/350-intersection-of-two-arrays-ii.cs
public class Solution {
    public int[] Intersect(int[] nums1, int[] nums2) {
        var dict = new Dictionary<int,int>();
        var intersection = new List<int>();
        foreach(var num in nums1){
            if(!dict.ContainsKey(num))
                dict.Add(num,1);
            else
                dict[num] += 1;
        }
        
         foreach(var num in nums2){
            if(dict.ContainsKey(num) && dict[num] != 0){
                dict[num] -= 1;
                intersection.Add(num);
            }
        }
        
        return intersection.ToArray();
    }
    
}

File: 350-intersection-of-two-arrays-ii/350-intersection-of-two-arrays-ii.ts
function intersect(nums1: number[], nums2: number[]): number[] {
    const intersection = [];
    let short, long;
    if(nums1.length < nums2.length) {
        short = nums1;
        long = nums2;
    } 
    else 
    {
        short = nums2;
        long = nums1;
    }
    
    for(const num of short){
        //Intersection
        if(short.includes(num) && long.includes(num)){
            intersection.push(num);
            const index = long.indexOf(num);
            long.splice(index,1);
        }
    }
    return intersection;
};